design/hardware-classification-controller/support-for-new-parameters-hwcc-DiskAndNIC.md

```diff
@@ -0,0 +1,227 @@
+   according to requirement. Then specify the disk type using combination
+   of `hctl` and `rotational` parameters using Rule Book.
+   Below YAML example indicates that user want 1 disk1 with pattern given in
+   `diskSelector` field with size of 2200 GB.
+
+     ```yaml
+     spec:
+     hardwareCharacteristics:
+       Disk:
+          minimumCount: 1
+          minimumIndividualSizeGB: 2200
+          diskSelector:
+            - hctl: "0:0:N:0"
+              rotational: false
+     ```
+
+   YAML rules for multiple Disks:
+
+   User can provide `minimumCount` and `minimumIndividualSizeGB`
+   according to requirement. Then specify the disk type using combination
+   of `hctl` and `rotational` parameters using Rule Book for multiple disks
+   as shown in example.
+   Below YAML example indicates that user want 4 disks with pattern given in
+   `diskSelector` field with individual size of 2200 GB.
+
+     ```yaml
+     spec:
+     hardwareCharacteristics:
+       Disk:
+          minimumCount: 4
+          minimumIndividualSizeGB: 2200
+          diskSelector:
+            - hctl: "0:0:N:0"
+              rotational: false
+            - hctl: "0:0:N:0"
+              rotational: true
+     ```
+
+1. `hardwareCharacteristics.nic.vendorSelector`
+
+   `model` field under `status.hardwareDetails.nics` contains
+   combination of Vendor ID and Product ID.
+
+   Vendor ID will be extracted from `status.hardwareDetails.nics.model`
+   and compared against new `vendor` fields under
+   `hardwareCharacteristics.nic.vendorSelector`.
+   Example: Intel's vendor ID is "0x8086".
+
+   User can provide multiple vendors under `vendorSelector`.
+   Dell supported NIC Vendor name and ID are present in rule book.
+
+   The classification of `vendor` will be done using exact match
+   of the values provided by user in profile.
+
+   Investigation details: Below is sample introspection data for NICs.
+
+      ```yaml
+     hardware:
+       nics:
+       - ip: ""
+         mac: 24:6e:96:aa:bb:cc
+         model: 0x8086 0x1234
+         name: eth3
+         pxe: false
+     ```
+
+### Implementation Details/Notes/Constraints
+
+Link for Existing HWCC Specs
+[Existing YAML](https://github.com/metal3-io/hardware-classification-controller/blob/master/config/samples/metal3.io_v1alpha1_hardwareclassification.yaml)
+
+* Below is sample yaml for additional parameters in HardwareClassification.
+
+   ```yaml
+
+   apiVersion: metal3.io.sigs.k8s.io/v1alpha1
+   kind: HardwareClassification
+   metadata:
+     name: profile1
+   spec:
+     hardwareCharacteristics:
+       Disk:
+          minimumCount: 5
+          minimumIndividualSizeGB: 2200
+          diskSelector:
+            - hctl: "0:0:N:0"
+              rotational: false
+            - hctl: "0:0:N:0"
+              rotational: true
+            - hctl: "0:N:0:0"
+              rotational: true
+            - hctl: "N:0:0:0"
+              rotational: false
```

Comment:
introducing diskSelector is good idea. It would be great if you can provide a table where it defines what's the meaning of each of these combination and how user will use it.

Reply:
Yes Digambar, we have rule book for all the `hctl` and `rotational` parameters. We will push that under docs folder of HWCC.
In Proposal highlighting it as Note.